### PR TITLE
fix: fixes textarea sizing

### PR DESCRIPTION
**Pull request by aradmargalit**
A better solution than https://github.com/eshedmargalit/paperaweek/pull/128. Try it out, let me know what you think!

### CODE REVIEW

File: client/src/components/MarkdownTextArea/useAutoResizingTextArea.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      onBlurHandler();
+    }
+
+    // Call our unfocus after, to switch back to the markdown render
+    setIsFocused(false);
+  };
+
+  const willRenderMarkdown = !isFocused && !!value && shouldRenderMarkdown;
+
+  useLayoutEffect(() => {
```
Comment (aradmargalit) [inline client/src/components/MarkdownTextArea/useAutoResizingTextArea.ts:33]: this is the real bread and butter:
1. Use `useLayoutEffect` to make sure updates to the DOM are made before React paints the DOM
2. Use the linked SO solution to get a (seemingly) perfect height calculation.
3. As a bonus, jump the cursor to the end

File: client/src/components/ReviewWizard/Form.tsx [hunk 2]
```diff
@@ -43,133 +43,141 @@ export default function Form({ initialReview, onChange, onSubmit }: FormProps):
     reValidateMode: 'onBlur',
   });
 
-  const convertAndSave = () => onChange(convertFormValuesToReview(getValues()));
+  const [lastSavedValues, setLastSavedValues] = useState(getValues());
+
+  const convertAndSave = () => {
+    // only save the draft if the new values are different from the old values
+    const currentValues = getValues();
+    if (!_isEqual(currentValues, lastSavedValues)) {
```
Comment (aradmargalit) [inline client/src/components/ReviewWizard/Form.tsx:51]: lots of ways to do this (like `JSON.stringify`) but I thought `_isEqual` was more expressive in terms of indicating that we care about deep equality

File: client/src/components/ReviewWizard/Form.tsx [hunk 3]
```diff
@@ -43,133 +43,141 @@ export default function Form({ initialReview, onChange, onSubmit }: FormProps):
...
+    }
+  };
+
   const convertAndSubmit = (formReview: FormReview) => onSubmit(convertFormValuesToReview(formReview));
 
   const reviewItemColSpan = { lg: 12, sm: 24 };
 
   return (
     <form>
-      <DraftsContext.Provider value={convertAndSave}>
```
Comment (aradmargalit) [inline client/src/components/ReviewWizard/Form.tsx:53]: the only change here is that I removed the unused `DraftsContextProvider`, the rest is just indented in, so Git marks it as changed. I promise I didn't touch it!

**Review by aradmargalit — COMMENTED** [review]
I removed the `DraftsContextProvider` since you weren't using it

**Review by eshedmargalit — APPROVED** [review]
This looks great! Works super well, and the cursor reliably jumping to the end without the (jarring) visual effect `onFocus` is a treat. 💯

**Pull request merged**